### PR TITLE
GHC 9.4 Support

### PR DESCRIPTION
**Pull request by parsonsmatt**
Before submitting your PR, check that you've:

- [x] Bumped the version number
- [x] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddock

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

File: warp/Network/Wai/Handler/Warp/Settings.hs [hunk 1]
```diff
@@ -249,7 +249,16 @@ exceptionResponseForDebug e =
 -- @since 3.3.17
 defaultFork :: ((forall a. IO a -> IO a) -> IO ()) -> IO ()
 defaultFork io =
+#if __GLASGOW_HASKELL__ >= 904
```
Comment (kazu-yamamoto) [inline warp/Network/Wai/Handler/Warp/Settings.hs:252]: Would you provide us a material to know the background of this change?

Comment (parsonsmatt) [reply warp/Network/Wai/Handler/Warp/Settings.hs:252]: Would love to but could not find any info on the changelog for `base` or `ghc-prim` about this. When GHC 9.4 is closer to release, I hope that this info will be more obvious.

Comment (kazu-yamamoto) [reply warp/Network/Wai/Handler/Warp/Settings.hs:252]: @snoyberg I trust Matt and have no objection for merging. What about you?

**Review by kazu-yamamoto — APPROVED** [review]
LGTM

File: warp/Network/Wai/Handler/Warp/Settings.hs (hunk 1 above)
Comment (snoyberg) [reply warp/Network/Wai/Handler/Warp/Settings.hs:252]: No objection from me either, go for it! As always thanks Matt 💪

Comment (parsonsmatt) [reply warp/Network/Wai/Handler/Warp/Settings.hs:252]: Of course!

True 9.4 support relies on updated `foundation` and `basement`, but I think we can release this as-is and it'll work when those libraries are up-to-date.

I can leave the `cabal.project` in the repo and merge as-is, or we can leave pending until that's been taken care of.

**Comment by kazu-yamamoto** [comment]
Merged and a new version has been released.
Thank you for your contribution!

**Pull request merged**